File: Selenium/Activities/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait


with webdriver.Firefox() as driver:

    driver.get("https://v1.training-support.net/selenium/dynamic-attributes")
    print("title is : ", driver.title)

    username = driver.find_element(By.XPATH, "//input[contains(@class, '-username')]")
    password = driver.find_element(By.XPATH, "//input[contains(@class, '-password')]")
    confirm_password = driver.find_element(By.XPATH, "//label[contains(text(), 'Confirm Password')]//following::input")

    email = driver.find_element(By.XPATH, "//input[contains(@class, 'email-')]")

    username.send_keys("MyUsername")
    password.send_keys("MyPassword2020!")
    confirm_password.send_keys("MyPassword2020!")
    email.send_keys("dummy@gmail.com")
    
    driver.find_element(By.XPATH, "//button[text()='Sign Up']").click()

    wait = WebDriverWait(driver, 10)
    wait.until(EC.visibility_of_element_located((By.ID, "action-confirmation")))
   
    message = driver.find_element(By.ID, "action-confirmation").text
    print("Success message : ", message)

    driver.quit()

File: Selenium/Activities/Activity16.java
package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity16 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://v1.training-support.net/selenium/dynamic-attributes");

		System.out.println("The Title of the page is  : " + driver.getTitle());

		WebElement userName = driver.findElement(By.xpath("//input[contains(@class, '-username')]"));
		WebElement password = driver.findElement(By.xpath("//input[contains(@class, '-password')]"));
		WebElement confirmPassword = driver
				.findElement(By.xpath("//label[text() = 'Confirm Password']/following-sibling::input"));
		WebElement email = driver.findElement(By.xpath("//label[contains(text(), 'mail')]/following-sibling::input"));

		userName.sendKeys("Tester");
		password.sendKeys("MyPassword");
		confirmPassword.sendKeys("MyPassword");
		email.sendKeys("dummy@gmail.com");

		driver.findElement(By.xpath("//button[contains(text(), 'Sign Up')]")).click();

		String loginMessage = driver.findElement(By.id("action-confirmation")).getText();
		System.out.println("success message: " + loginMessage);

		driver.quit();

	}

}
